test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go

```diff
@@ -666,7 +666,183 @@ var _ = f.CLICoreDescribe("[Tests:E2E][Feature:Plugin-Sync-TMC-lifecycle]", func
 		})
 	})
 
-	// Use Case 7: Plugin List, sync, search and install functionalities with Context Issues
+	// Use case 7: Sync for single target specific plugins, and validate the plugin list
+	// run context create (make sure another target context is active, but yet to install plugins), it should not perform the sync for all active contexts
+	// run target specific plugin sync (for k8s target), make sync should not happen for tmc context even though its active
+	// run target specific plugin sync (for tmc target), make sync should not happen for k8s context even though its active
+	Context("Use case: create k8s and tmc specific contexts, validate plugins list and perform pluin sync, and perform context switch", func() {
+		var clusterInfo *f.ClusterInfo
+		var pluginCRFilePaths []string
+		var pluginsInfoForCRsApplied, installedPluginsListK8s []*f.PluginInfo
+		var contextNameK8s string
+		contexts := make([]string, 0)
+		totalInstalledPlugins := 1 // telemetry plugin that is part of essentials plugin group will always be installed
+		var err error
+		// Test case: a. k8s: create KIND cluster, apply CRD
+		It("create KIND cluster", func() {
+			// Create KIND cluster, which is used in test cases to create context's
+			clusterInfo, err = f.CreateKindCluster(tf, f.ContextPrefixK8s+f.RandomNumber(4))
+			Expect(err).To(BeNil(), "should not get any error for KIND cluster creation")
+		})
+		// Test case: b. k8s: apply CRD (cluster resource definition) and CR's (cluster resource) for few plugins
+		It("apply CRD and CRs to KIND cluster", func() {
+			err = f.ApplyConfigOnKindCluster(tf, clusterInfo, append(make([]string, 0), f.K8SCRDFilePath))
+			Expect(err).To(BeNil(), "should not get any error for config apply")
+
+			pluginsToGenerateCRs, ok := pluginGroupToPluginListMap[usePluginsFromK8sPluginGroup]
+			Expect(ok).To(BeTrue(), "plugin group is not exist in the map")
+			Expect(len(pluginsToGenerateCRs) > numberOfPluginsToInstall).To(BeTrue(), "we don't have enough plugins in local test central repo")
+			pluginsInfoForCRsApplied, pluginCRFilePaths, err = f.CreateTemporaryCRsFromPluginInfos(pluginsToGenerateCRs[:numberOfPluginsToInstall])
+			Expect(err).To(BeNil(), "should not get any error while generating CR files")
+			err = f.ApplyConfigOnKindCluster(tf, clusterInfo, pluginCRFilePaths)
+			Expect(err).To(BeNil(), "should not get any error for config apply")
+			totalInstalledPlugins += numberOfPluginsToInstall
+		})
+
+		// Test case: c. k8s: create context and make sure context has created
+		It("create context with kubeconfig and context", func() {
+			By("create context with kubeconfig and context")
+			contextNameK8s = f.ContextPrefixK8s + f.RandomString(4)
+			err := tf.ContextCmd.CreateContextWithKubeconfig(contextNameK8s, clusterInfo.KubeConfigPath, clusterInfo.ClusterKubeContext)
+			Expect(err).To(BeNil(), "context should create without any error")
+			active, err := tf.ContextCmd.GetActiveContext(string(types.TargetK8s))
```

Comment:
Should `GetActiveContext` take a contextType as a parameter?

Reply:
updated